File: src/rules/cors.py
from flask import Flask
from flask_cors import CORS

app = Flask(__name__)

# ruleid: patronus-cors-wildcard-origin-python
CORS(app, origins="*")

# ruleid: patronus-cors-wildcard-origin-python
CORS(app)


def add_header(resp):
    # ruleid: patronus-cors-wildcard-origin-python
    resp.headers["Access-Control-Allow-Origin"] = "*"
    return resp


# ok: patronus-cors-wildcard-origin-python
CORS(app, origins=["https://app.example.com"])


def safe_header(resp):
    # ok: patronus-cors-wildcard-origin-python
    resp.headers["Access-Control-Allow-Origin"] = "https://app.example.com"
    return resp

File: src/rules/cors.js
const cors = require('cors');
const express = require('express');
const app = express();

function setup(res) {
  // ruleid: patronus-cors-wildcard-origin
  res.header('Access-Control-Allow-Origin', '*');

  // ruleid: patronus-cors-wildcard-origin
  res.setHeader('Access-Control-Allow-Origin', '*');
}

// ruleid: patronus-cors-wildcard-origin
app.use(cors({ origin: '*' }));

// ruleid: patronus-cors-wildcard-origin
app.use(cors({ origin: true, credentials: true }));

// ok: patronus-cors-wildcard-origin
app.use(cors({ origin: ['https://app.example.com'] }));

function safe(res) {
  // ok: patronus-cors-wildcard-origin
  res.header('Access-Control-Allow-Origin', 'https://app.example.com');
}
